contracts/AssetHolder.sol

```diff
@@ -13,7 +13,7 @@ contract AssetHolder {
         // Moreover, the participant should sign the address that they wish
         // to send the transaction from, preventing any replay attack.
         address participant; // the account used to sign commitment transitions
-        bytes32 destination; // either an account or a channel
+        address destination; // either an account or a channel
```

Comment:
Is this right?

Reply:
For code relating to withdrawals, I chose to use `address` as much as possible, rather than using `bytes32` and having to convert it to an `address` to do the actual payout (we cannot payout to anything other than an `address payable`). So when you submit a `withdraw` tx, you specify an EOA or contract to withdraw to, and the `AssetHolder` will convert it to the correct format for nitro's internal bookkeeping (e.g. to make relevant checks). 

The alternative is to have clients format withdrawal addresses with the trailing zeros, and the AssetHolder strips them out before doing the payout. This is how it works for `deposit`. To me the asymmetry in the current solution reflects that you can deposit into a contract or a participant, but you can only withdraw from an external address.

It should probably *not* say `// either an account or a channel`. Will change.

Reply:
Correction -- got a bit confused there between `participant` and `destination`. 

`participant` needs to be represented as `bytes32` for checks and effects, and I suppose I have made the choice that the contract pads it out with zeros. It would save a bit of gas to have the client do the padding. 

`destination` has to be an `address` because it is where ETH or tokens get sent.